src/Context/BreweryContext.js

```diff
@@ -16,93 +19,72 @@ export function BreweryContextProvider({ children }) {
         return { ...state, selectedBrewery: action.id, isSelected: true };
       case 'SET_IS_SELECTED':
         return { ...state, isSelected: action.status };
+      case 'SET_BREWERIES':
+        return { ...state, breweries: action.breweries };
+      case 'SET_NO_RESULTS':
+        return { ...state, noResults: action.noResults }
+      case 'SET_ERROR':
+        return {...state, error: action.error}
       default:
         return state;
     }
   };
 
   const [state, dispatch] = useReducer(breweryReducer, initialState);
 
-  const [breweries, setBreweries] = useState([]);
-  const [noResults, setNoResults] = useState(false);
-  const [error, setError] = useState('');
+  function cleanData(data, city, state) {
+    let validData = data.filter(
+      brewery => brewery.latitude && brewery.longitude,
+    );
 
-  // const [selectedBrewery, setSelectedBrewery] = useState({}); 
-  // const [isSelected, setIsSelected] = useState(false)
-
-  async function obtainBreweries(city, state) {
-    let stateBreweryData = [];
-    let breweryData = [];
-    let filteredBreweryData = [];
-
-    switch (city) {
-      case '':
-        stateBreweryData = await getBreweriesByState(state);
-
-        if (stateBreweryData.name !== 'Error') {
-      let onlyCoordsData = stateBreweryData.filter(brewery => brewery.longitude && brewery.latitude)
-          setError(false);
-          setBreweries(onlyCoordsData);
-          setBreweries(onlyCoordsData);
-          setNoResults(false);
-        }
-        // Set error by default
-        setError(stateBreweryData.message);
-
-        break;
-
-      default:
-        breweryData = await getBreweriesByCity(city);
-
-        if (breweryData.name === 'Error') {
-          setError(breweryData.message);
-          return;
-        }
+    if (city) {
+      validData = validData.filter(brewery => brewery.state === state);
+    }
 
-        setError(false);
-        filteredBreweryData = breweryData.filter(
-          brewery => brewery.state === state && brewery.latitude && brewery.longitude
-        );
+    return validData;
+  }
 
-        if (filteredBreweryData.length === 0) {
-          setNoResults(true);
-          setBreweries(filteredBreweryData);
-        } else {
-          setNoResults(false);
-          setBreweries(filteredBreweryData);
-        }
-    }
+  function processData(breweries) {
+    const noResults = breweries.length ? false : true;    
```

Comment:
Small tweak I would like to suggest but wouldn't request changes for, unless you would like to take the time, is to leverage JS's truthy/falsy behavior here and on other ternaries in this PR. You can achieve this using the bang operator. (i.e. !breweries.length) Will return true if 0 and false if not 0.